tests/bson/bson-fromPHP-002.phpt

```diff
@@ -0,0 +1,55 @@
+--TEST--
+BSON\fromPHP(): Encoding non-Persistable objects as a document
+--INI--
+date.timezone=America/Los_Angeles
+--SKIPIF--
+<?php require __DIR__ . "/../utils/basic-skipif.inc"?>
+--FILE--
+<?php
+use MongoDB\BSON as BSON;
+
+require_once __DIR__ . "/../utils/basic.inc";
+
+class MyDocument {
+    private $foo = 1;
+    protected $bar = 2;
+    public $baz = 3;
+}
+
+$utcdatetime = new BSON\UTCDateTime("1416445411987");
+
+$datetime = $utcdatetime->toDateTime();
+var_dump($datetime->format(DATE_RSS));
+var_dump((string) $utcdatetime);
+
+$tests = array(
+    $utcdatetime,
+    $datetime,
+    new MyDocument,
+);
+
+foreach ($tests as $document) {
+    $s = fromPHP($document);
+    echo "Test ", toJSON($s), "\n";
+    hex_dump($s);
+}
+
+?>
+===DONE===
+<?php exit(0); ?>
+--EXPECT--
+string(31) "Thu, 20 Nov 2014 01:03:31 +0000"
+string(13) "1416445411987"
+Test { }
+     0 : 05 00 00 00 00                                   [.....]
+Test { "date" : "2014-11-20 01:03:31.987000000", "timezone_type" : 1, "timezone" : "+00:00" }
+     0 : 55 00 00 00 02 64 61 74 65 00 1e 00 00 00 32 30  [U....date.....20]
+    10 : 31 34 2d 31 31 2d 32 30 20 30 31 3a 30 33 3a 33  [14-11-20 01:03:3]
+    20 : 31 2e 39 38 37 30 30 30 30 30 30 00 10 74 69 6d  [1.987000000..tim]
+    30 : 65 7a 6f 6e 65 5f 74 79 70 65 00 01 00 00 00 02  [ezone_type......]
+    40 : 74 69 6d 65 7a 6f 6e 65 00 07 00 00 00 2b 30 30  [timezone.....+00]
+    50 : 3a 30 30 00 00                                   [:00..]
+Test { "foo" : 1, "bar" : 2, "baz" : 3 }
```

Comment:
This will most certainly change once we address [PHPC-393](https://jira.mongodb.org/browse/PHPC-393) and stop serializing private/protected properties.